src/DependencyInjection/Compiler/ObjectAclManipulatorCompilerPass.php

```diff
@@ -30,7 +30,18 @@ public function process(ContainerBuilder $container): void
         $availableManagers = [];
 
         foreach ($container->getServiceIds() as $id) {
-            if (0 !== strpos($id, 'sonata.admin.manipulator.acl.object.') || !is_subclass_of($container->getDefinition($id)->getClass(), ObjectAclManipulatorInterface::class)) {
+            if (0 !== strpos($id, 'sonata.admin.manipulator.acl.object.') || null === $class = $container->getDefinition($id)->getClass()) {
+                continue;
+            }
+
+            // We trim the possible "%" characters around the class definition since it could be using "%parameter%" syntax.
+            $class = trim($class, '%');
+
+            if (!class_exists($class, false) && $container->hasParameter(str_replace('%', '', $class))) {
```

Comment:
Character "%" is already trimmed at line 38.
```suggestion
            if (!class_exists($class, false) && $container->hasParameter($class)) {
```